specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2020-05-15/timeseriesinsights.json

```diff
@@ -1406,7 +1406,21 @@
     },
     "EnvironmentUpdateParameters": {
       "type": "object",
+      "discriminator": "kind",
```

Comment:
Add this to discriminate "Gen1EnvironmentUpdateParameters" and "Gen2EnvironmentUpdateParameters".

Reply:
`kind` needs to be annotated as `required`.

Reply:
> Discriminator - `kind` needs to be annotated as `required`.

But it will not in request body.

Reply:
@kairu-ms , can you confirm if service doesn't require `kind`?

Reply:
Yes. I've hade a request which is failed with `kind` property.